### ISSUE TITLE
crash when reloading in simulator "pointerValue"

### ISSUE THREAD

**Issue by grittathh** [andpor/react-native-sqlite-storage#58]
With the latest commit, reloading the simulator with an already-loaded DB causes a crash. The stack trace hints that the error is in dealloc (in SQLite.m) and has to do with pointerValue being unrecognized.

This seems to fix it:

```
-(void)dealloc
{
  int i;
  NSArray *keys = [openDBs allKeys];
  NSDictionary *dbInfo;
  NSString *key;
  sqlite3 *db;

  /* close db the user forgot */
  for (i=0; i<[keys count]; i++) {
    key = [keys objectAtIndex:i];
    dbInfo = [openDBs objectForKey:key];
    db = [((NSValue *) dbInfo[@"dbPointer"]) pointerValue];
    sqlite3_close (db);
  }

#if !__has_feature(objc_arc)
  [openDBs release];
  [appDBPaths release];
  [super dealloc];
#endif
}
```

**Comment by andpor** [comment]
it should be fixed. Can you give it a quick test?

**Comment by andpor** [comment]
resolved in latest 3.0 build
